---
title: "Lecture 13"
author: "Cassandra Chow"
date: "May 15, 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##1000 Genomes Data Analysis

Downloaded CSV file for MXL 
```{r}
#Read CSV 
MXL <- read.csv("373531-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv", header = TRUE)
```

Finding the percentage
```{r}
MXL.summary <- table(MXL[,2])
MXL.summary/nrow(MXL)*100
```

The answer is 14% (of G|G)

###Side-note: Fasta quality scores 
```{r}
library(seqinr)
library(gtools)
phred <- asc(s2c("DDDDCDEDCDDDDBBDDDCC@")) - 33 
phred
```

```{r}
prob <- 10**(-phred/10)
prob
```


##Population Scale Analysis 

Read this file 
```{r}
txt <- read.table("rs8067378_ENSG00000172057.6.txt")
summary(txt)
```

Looking at the data, also seeing the amount of patients
```{r}
table(txt$geno)

paste("The number of patients is", nrow(txt))
```

Set numeric values
```{r}
boxplot(exp ~ geno, data = txt, notch = TRUE)
```
You can see that there's no overlap, so these SNPs are significant 

Install ggplot
```{r}
library(ggplot2)
```

```{r}
ggplot(txt, aes(geno, exp)) + geom_boxplot()
```

